### PR TITLE
feat(website): add Git Hooks recipe

### LINKED ISSUE #1198: 📎 Create a new `git hooks` recipe
### Description

Essentially, on this page we want to add some recipes for how to add Biome to commit hooks:
- lint-staged
- lefthook
- etc.

@nhedger:  I know you set up a commit hook in `biome-vscode`, this knowledge could live in that page
@Conaclos : I know you created a repository around this, but honestly, I don't know what that is about. We can have some knowledge in that page

The name of the page isn't set on stone, feel free to come up with a better name

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Fix #1198

Add Git Hooks page

### CODE REVIEW

**Comment by nhedger** [comment]
Added instructions for lefthook

**Comment by ematipico** [comment]
I added the instructions for `lint-staged`

File: website/src/content/docs/recipes/git-hooks.mdx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+description: Using Biome in Git Hooks
+---
+
+Git allows executing scripts during the run of a git command using [Git Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks).
+For example, you can format and lint the staged files before committing or pushing.
+Several tools exist to simplify the management of Git Hooks.
+In the following sections we introduce some of them and how they can be used with Biome.
+
+
+## lefthook
```
Comment (nhedger) [inline website/src/content/docs/recipes/git-hooks.mdx:12]:
```suggestion
## Lefthook
```

File: website/src/content/docs/recipes/git-hooks.mdx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+[lefthook](https://github.com/evilmartians/lefthook) provides a fast, cross-platform, and dependency-free hook manager.
```
Comment (nhedger) [inline website/src/content/docs/recipes/git-hooks.mdx:14]:
```suggestion
[Lefthook](https://github.com/evilmartians/lefthook) provides a fast, cross-platform, and dependency-free hook manager.
```

File: website/src/content/docs/recipes/git-hooks.mdx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+It can be [installed via NPM](https://github.com/evilmartians/lefthook#install).
+
+Add a file named `lefthook.yml` at the root of your Git repository.
+Some examples of _lefthook_ configurations:
```
Comment (nhedger) [inline website/src/content/docs/recipes/git-hooks.mdx:18]:
```suggestion
Some examples of _Lefthook_ configurations:
```

File: website/src/content/docs/recipes/git-hooks.mdx [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+Note that you don't need to use both `glob` and `--files-ignore-unknown=true`.
+Using only `--files-ignore-unknown=true` allows handling files supported in the present and in the future by Biome.
+If you wish more control over which files are handled, you should use `glob`.
+
+`--no-errors-on-unmatched` silents possible errors in case *no files are processed*.
+
+Once configured, run `lefthook install` to set up the hooks.
+
+If your project contains a `package.json`,
+you can automatically set up _lefthook_ hooks upon package installation using `scripts.postinstall`:
```
Comment (nhedger) [inline website/src/content/docs/recipes/git-hooks.mdx:61]:
```suggestion
you can automatically set up _Lefthook_ hooks upon package installation using `scripts.postinstall`:
```

**Review by ematipico — APPROVED** [review]

**Pull request merged**